pandas/tests/indexes/test_base.py

```diff
@@ -1661,9 +1662,13 @@ def test_isin_nan_common_object(self, nulls_fixture, nulls_fixture2):
         # Test cartesian product of null fixtures and ensure that we don't
         # mangle the various types (save a corner case with PyPy)
 
-        if PYPY and nulls_fixture is np.nan:  # np.nan is float('nan') on PyPy
+        # all nans are the same
```

Comment:
so this test seems now very restrictive?

Reply:
did you remove the PYPY on purpose?

Reply:
cc @mattip can you comment on the equivalence of nulls here? we don't test on PYPY, can you see if this still fails?

Reply:
@jreback After the fix the behavior is less subtle:

``` 
Index([nulls_fixture]).isin([nulls_fixture2])
```

is always `True` as long as `nulls_fixture` and `nulls_fixture2`  are nans, no matter whether those are the same nan-objects or different nan-objects. Thus PyPy is no longer a special case. But I must confess, I didn't test with PyPy and assumed this would be done somewhere in CI.

It seems as if the purpose of this test is to check, that the identity of the object is preserved (something similar to bug #22160). In this case `np.nan`  is no longer a good tool to do this check, because `isin`  bcomes agnostic to the exact identity of the nan-object.

Reply:
this particular test in its 25 variants passes on PyPy. There are other failures but they seem unrelated.